lms/djangoapps/courseware/courses.py

```diff
@@ -113,21 +113,21 @@ def check_course_access(course, user, action, check_if_enrolled=False):
     Check that the user has the access to perform the specified action
     on the course (CourseDescriptor|CourseOverview).
 
-    check_if_enrolled: If true, additionally verifies that the user is either
-    enrolled in the course or has staff access.
+    check_if_enrolled: If true, additionally verifies that the user is enrolled.
     """
-    access_response = has_access(user, action, course, course.id)
+    # Allow staff full access to the course even if not enrolled
+    if has_access(user, 'staff', course.id):
```

Comment:
Good questions:

1. My understanding is that this isn't changing the logic, because `has_access` would return true for staff. It just moves the staff check earlier to make it clearer (to my mind) that staff gets to skip all the other logic.

2. I'm not sure what you're saying. Are you thinking there's a performance impact to moving the staff check earlier? That probably is why some of the counts increased, so we could optimize it. However, I think that the clarity of understanding that staff always get access is a worthwhile trade off.